cmd/lncli/invoicesrpc_active.go

```diff
@@ -51,29 +54,21 @@ var settleInvoiceCommand = cli.Command{
 }
 
 func settleInvoice(ctx *cli.Context) error {
-	var (
-		preimage []byte
-		err      error
-	)
-
 	client, cleanUp := getInvoicesClient(ctx)
 	defer cleanUp()
 
-	args := ctx.Args()
-
-	switch {
-	case ctx.IsSet("preimage"):
-		preimage, err = hex.DecodeString(ctx.String("preimage"))
-	case args.Present():
-		preimage, err = hex.DecodeString(args.First())
+	hash, err := hex.DecodeString(ctx.String("hash"))
```

Comment:
why do the user need to supply both? if the preimage is being supplied externally we can calculate the hash

Reply:
fixed

Reply:
Ah, it didn't need fixing, because if hash isn't set, an empty slice will be passed and the server will deal with it. Reverted fix